x/claim/keeper/invariants.go

```diff
@@ -0,0 +1,82 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ignite/modules/x/claim/types"
+)
+
+const (
+	airdropSupplyRoute = "airdrop-supply"
+)
+
+// RegisterInvariants registers all module invariants
+func RegisterInvariants(ir sdk.InvariantRegistry, k Keeper) {
+	ir.RegisterRoute(types.ModuleName, airdropSupplyRoute,
+		AirdropSupplyInvariant(k))
+}
+
+// AllInvariants runs all invariants of the module.
+func AllInvariants(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		return AirdropSupplyInvariant(k)(ctx)
+	}
+}
+
+// AirdropSupplyInvariant invariant checks that airdrop supply is equal to the remaining claimable
+// amounts in claim records
+func AirdropSupplyInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+
+		// check missions
+		missionMap := make(map[uint64]types.Mission)
+		missions := k.GetAllMission(ctx)
+
+		for _, mission := range missions {
+			missionMap[mission.MissionID] = mission
+		}
+
+		// check claim records
+		claimSum := sdkmath.ZeroInt()
+		claimRecordMap := make(map[string]struct{})
+		claimRecords := k.GetAllClaimRecord(ctx)
+
+		for _, claimRecord := range claimRecords {
```

Comment:
Maybe it can be possible to manage duplicated logic with airdrop supply verification in `types/genesis.go`?

With a method like
```
func CheckAirdropSupply(airdropSupply sdk.Coins, m []Mission, c []ClaimRecord) error
```